mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -744,6 +747,42 @@ impl S3CrtClientInner {
     }
 }
 
+/// S3 operation supported by this client.
+#[derive(Debug, Clone, Copy)]
+enum S3Operation {
+    DeleteObject,
+    GetObject,
+    GetObjectAttributes,
+    HeadBucket,
+    HeadObject,
+    ListObjects,
+    PutObject,
+}
+
+impl S3Operation {
+    /// The [MetaRequestType] to use for this operation.
+    fn meta_request_type(&self) -> MetaRequestType {
+        match self {
+            S3Operation::GetObject => MetaRequestType::GetObject,
+            S3Operation::PutObject => MetaRequestType::PutObject,
+            _ => MetaRequestType::Default,
+        }
+    }
+
+    /// The operation name to set when configuring a request, if required.
+    fn operation_name(&self) -> Option<&'static str> {
+        match self {
+            S3Operation::DeleteObject => Some("DeleteObject"),
+            S3Operation::GetObject => None,
```

Comment:
Is there any harm to always return operation name here even though it's not always required?

Reply:
The CRT docs only say it's required for Default meta requests. It's likely ignored in other cases, but it seems sensible to leave it unset: e.g. which operation would we use for a "PutObject" meta-request, which may translate into CreateMPU, UploadPart, etc.?

Reply:
>which operation would we use for a "PutObject" meta-request

This is transparent to us, the users of CRT. So, it should not matter. If CRT decides to split an operation, it is their business.

I don't think this is a blocking thing, but I can see how it is more simple to always return and set an operation name.

Reply:
(Related to the unit test question) How do you stick to their guidance without a test that fails if we anyone intends to send an operation name for GET or PUT?

Reply:
@arsh, maybe I am missing something. What kind of test do you have in mind?

Reply:
For example, if you think we should not be sending an operation name for PUT and GET, then a unit test can help enforce that. With those tests, you are codifying your current assumption until those change. Without any tests, I can start returning an operation name for those any time and no test will fail.